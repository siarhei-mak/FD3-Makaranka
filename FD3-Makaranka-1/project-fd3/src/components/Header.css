.Header{
  background-color: rgb(85, 72, 5);
  /* border: 1px solid silver; */
  /*height: 15vh;*/
  box-sizing: border-box;
  display: flex;
flex-direction: column;
align-content: flex-end;
justify-content: flex-end;
min-width: 360px;
}
.Header .banner_title_shop{
  display: inline-block;
  font-size: 30px;
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  color: rgb(154, 223, 27);
  padding:0px 0px;

}

.Header .HeaderCategory{
  width: 100%;
 align-self: flex-end;
}
