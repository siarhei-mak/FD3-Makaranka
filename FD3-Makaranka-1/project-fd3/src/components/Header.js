import React from 'react';
import './Header.css';

import { NavLink } from 'react-router-dom';
import HeaderCategory from './HeaderCategory'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });



let totalCategories=[
  {name:'Светлое пиво', id: 1, category:'lightBeer'},
  {name:'Темное пиво', id: 2, category:'darkBeer' },
  {name:'Крафтовое пиво', id: 3, category:'kraftBeer', },
  {name:'Пиво по фильмам', id: 4, category:'filmBeer',},
  {name:'Пивной сет', id: 5, category:'setBeer',},
  {name:'Закуска', id: 6, category:'snack'},

]

class Header extends React.Component{


  render(){
    return (
      <div className='Header'>

          <NavLink to="/" exact className="PageLink" activeClassName="ActivePageLink">
             <div className='banner_title_shop'>
              BEER-bar
             </div>
          </NavLink>


          <div className='HeaderCategory'>
          <HeaderCategory  categories={totalCategories} />

            </div>

      </div>
    )
  }

}

export default Header
