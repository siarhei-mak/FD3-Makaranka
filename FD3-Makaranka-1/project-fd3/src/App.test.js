import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Header from '../src/components/Header'





describe('Test HEADER', () => { 

  it('with shallow HEADER', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();


  })


})
