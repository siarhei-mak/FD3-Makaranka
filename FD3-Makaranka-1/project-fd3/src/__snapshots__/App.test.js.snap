// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HEADER with shallow HEADER 1`] = `
<div
  className="Header"
>
  <NavLink
    activeClassName="ActivePageLink"
    aria-current="page"
    className="PageLink"
    exact={true}
    to="/"
  >
    <div
      className="banner_title_shop"
    >
      BEER-bar
    </div>
  </NavLink>
  <div
    className="HeaderCategory"
  >
    <HeaderCategory
      categories={
        Array [
          Object {
            "category": "lightBeer",
            "id": 1,
            "name": "Светлое пиво",
          },
          Object {
            "category": "darkBeer",
            "id": 2,
            "name": "Темное пиво",
          },
          Object {
            "category": "kraftBeer",
            "id": 3,
            "name": "Крафтовое пиво",
          },
          Object {
            "category": "filmBeer",
            "id": 4,
            "name": "Пиво по фильмам",
          },
          Object {
            "category": "setBeer",
            "id": 5,
            "name": "Пивной сет",
          },
          Object {
            "category": "snack",
            "id": 6,
            "name": "Закуска",
          },
        ]
      }
    />
  </div>
</div>
`;
